### ISSUE TITLE
Kafka: performance regression in 19.15

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#7261]
We need to improve kafka consuming performance, degraded after #6950 (which is very important and desired change in all other aspects).

**Comment by josepowera** [comment]
We have seen big drop in performance after this upgrade (19.14 to 19.15). 

Since data consistency is more important (and we are really happy with progress in kafka support)  we postponed this question (we didn't know if there was any bigger change in our virtual machines provider)

However..  at the moment we are looking into:
-setting "kafka_num_consumers" in kafka table(up to # of partitions)
-max_insert_block_size   in configuration

What's the best temporary solution?

**Comment by filimonov** [comment]
Most probably both options will not really help. You can try to create one more (or few more tables) with the same structure filling the same target table (in that case it will work multithreaded, with num_consumers you are still locked in one thread)

**Comment by josepowera** [comment]
Has anyone benchmarked if #7475 makes a big difference?

**Comment by josepowera** [comment]
Ouch... I just hopped this was THE FIX.    Our CH is lagging behind and cleaning LAGs during offpeak- nights.

Has anyone tried performance of alternatives like:
-Kafka Connect JDBC with CSV->JDBC-> CH?
-sinker project (https://github.com/housepower/clickhouse_sinker)

@abyss7   Is there any solution planned for 19.17?

**Comment by simpl1g** [comment]
Do you have any updates on this issue?

**Comment by abyss7** [comment]
Will try to solve it in #7837
